src/FileIO.jl

```diff
@@ -0,0 +1,240 @@
+    write(file, entryname * "/VERSION", 0)
+    write(file, entryname * "/type", typeof(model))
+    write(file, entryname * "/data", model)
+    nothing
+end
+
+#     load_model(data, ::Type{Model})
+#
+# Loads a model from a given `data` dictionary produced by `JLD.load(filename)`.
+# The second argument can be used for dispatch between different models.
+function load_model(data, ::DataType)
+    @assert data["VERSION"] == 0
+    data["data"]
+end
+
+
+# TODO: Not used currently
+#     save_lattice(filename, lattice, entryname)
+#
+# Save (minimal) information necessary to reconstruct the given `lattice` in a
+# jld-file `filename` under group `entryname`.
+#
+# By default the full lattice object is saved. When saving a simulation, the
+# entryname defaults to `MC/Model/Lattice`.
+function save_lattice(filename::String, lattice::AbstractLattice, entryname::String)
+    mode = isfile(filename) ? "r+" : "w"
+    file = jldopen(filename, mode)
+    save_lattice(file, lattice, entryname)
+    close(file)
+    nothing
+end
+function save_lattice(file::JLD.JldFile, lattice::AbstractLattice, entryname::String)
+    write(file, entryname * "/VERSION", 0)
+    write(file, entryname * "/type", typeof(lattice))
+    write(file, entryname * "/data", lattice)
+    nothing
+end
+
+#     load_lattice(data, ::Type{Lattice})
+#
+# Loads a lattice from a given `data` dictionary produced by `JLD.load(filename)`.
+function load_lattice(data, ::DataType)
+    @assert data["VERSION"] == 0
+    data["data"]
+end
+
+
+@eval begin
+    """
+        save_rng(filename [; rng::MersenneTwister = GLOBAL_RNG, entryname="RNG"])
+
+    Saves the current state of Julia's random generator (`Random.GLOBAL_RNG`) to the
+    given `filename`.
+    """
+    function save_rng(
+            filename::String;
+            rng::MersenneTwister = $(VERSION < v"1.3.0" ? Random.GLOBAL_RNG : :(Random.default_rng())),
+            entryname::String="RNG"
+        )
+        mode = isfile(filename) ? "r+" : "w"
+        file = jldopen(filename, mode)
+        save_rng(file, rng=rng, entryname=entryname)
+        close(file)
+    end
+    function save_rng(
+            file::JLD.JldFile;
+            rng::MersenneTwister = $(VERSION < v"1.3.0" ? Random.GLOBAL_RNG : :(Random.default_rng())),
+            entryname::String="RNG"
+        )
+        try
+            write(file, entryname * "/seed", rng.seed)
+            write(file, entryname * "/state_val", rng.state.val)
+            write(file, entryname * "/vals", rng.vals)
+            write(file, entryname * "/ints", Int.(rng.ints))
+            write(file, entryname * "/idxF", rng.idxF)
+            write(file, entryname * "/idxI", rng.idxI)
+        catch e
+            error("Error while saving RNG state: ", e)
+        end
+    end
+
+    """
+      load_rng!(data[; rng=Random.GLOBAL_RNG, entryname="RNG"])
+
+    Loads an RNG from a given `data` dictinary generated by `JLD.load()` to `rng`.
+    """
+    function load_rng!(
```

Comment:
A nicer way to do this seems to be just `copy!(Random.default_rng(), loaded_rng);` where, for testing, `loaded_rng = Random.MersenneTwister()`.